package/base-files/files/lib/upgrade/stage2

```diff
@@ -76,40 +76,56 @@ switch_to_ramfs() {
 	}
 }
 
+get_remaining() {
+	local stat
+
+	local proc_pid proc_name proc_state proc_ppid proc_rest
+	read proc_pid proc_name proc_state proc_ppid proc_rest < /proc/$$/stat
+
+	for stat in /proc/[0-9]*/stat; do
+		[ -f "$stat" ] || continue
+
+		local pid name state ppid rest
+		read pid name state ppid rest < $stat
+		name="${name#(}"; name="${name%)}"
+
+		# Skip PID1, our parent, ourself and our children
+		[ $pid -ne 1 -a $pid -ne $proc_ppid -a $pid -ne $$ -a $ppid -ne $$ ] || continue
```

Comment:
```
[ "$ppid" = "I"  ] && continue
```

Reply:
sorry I don't get it, is that a typo?

Reply:
sh out of range?

Reply:
`$ppid` is a number and `I` is a letter....

Reply:
exactly

Reply:
note that I copied this line from below

this conditional needs to pass for some processes and fail for others, if I match ppid to a letter, it will either always pass or always fail....

Reply:
Maybe you assign some variables differently...? Apart from that, I'd not touch this copied code unless necessary due to the copying itself.

Reply:
I see nothing wrong with the line as it is, it works

Reply:
may work for you... but not for me... your pr... your call

i've patched this locally as i'm not a fan of errors during
the sysupgrade process....
```
#!/bin/sh

for stat in /proc/[0-9]*/stat; do
	[ -f "$stat" ] || continue
	read pid name state ppid rest < $stat 2>/dev/null    
        name="${name#(}"; name="${name%)}"
	echo "ppid:$ppid name:$name"
done

#######################################
[snip]
ppid:1 name:logd
ppid:1 name:rpcd
ppid:1 name:dropbear
ppid:1 name:netifd
ppid:1 name:odhcpd
ppid:1 name:crond
ppid:1 name:ubusd
ppid:4582 name:udhcpc
ppid:4582 name:odhcp6c
ppid:2 name:kblockd
ppid:1 name:askfirst
ppid:1 name:rngd
ppid:2 name:irq/39-aerdrv
ppid:2 name:hwrng
ppid:2 name:iscsi_eh
ppid:2 name:kworker/3:2H
ppid:I name:DWC
ppid:2 name:uas
ppid:2 name:kworker/u8:0-events_unbound
ppid:2 name:vchiq-slot/0
ppid:2 name:ipv6_addrconf
[snip]
##########################################
```

Reply:
what's not working for you?

also, the read command will not error because the previous line checks that the input file exists, and otherwise skips it

Reply:
some kernel process contain spaces...

while the read command does not error
the numerical comparison as mentioned
does with the phrase "sh out of range"
when it encounters the "I" char sucked
in by read and its failure to handle spaces
in "value 2" from stat

as we won't be killing kernel threads
anyway... I see no need to implement
more complex workarounds

continuing on the "I" char prior to the
numerical comparisons handle this
case well

Reply:
ok I understand now

do you have an example of a process with a name, so I can test?

Reply:
I think the solution is to not strip the parenthesis from the stat file, rather replace with quotes?

Reply:
the space would still cause read to split it...hmmm

Reply:
its and edge case... but the dd redirect shennanigans do provide
precendent for minimal false errors and crap during sysupgrade...

i'm not sure of the offender in my case "DWC Notifcation" is even
compliant with standards...

would be interesting to post the sample above on the forum and see
just how many people / procs match this...

for clean handling... (if seen as desired) 'shift on alpha' may work
fairly simplistically...

Reply:
@wulfy23 see the new commit and try it out 👍🏼

Reply:
damn... this implementation for handling
spaces in stat name has passed the
wulfy23 nitpick stresstest

```
ppid:2 name:DWC_Notificatio
```
;)

Reply:
Uhm who would have a whitespace in the process name? I don't think this is really required sorry...

Reply:
It would be best to correct that one program to not have a space in its name...

@wulfy23 where did it come from?

Reply:
the kernel: drivers/usb/host/dwc_common_port/dwc_notifier.c